### ISSUE TITLE
TextGenerationEnv cannot pass example metadata to reward functions

### ISSUE THREAD

**Issue by teilomillet** [teilomillet/textpolicy#2]
## Problem

`TextGenerationEnv` hardcodes `example={}` when calling reward functions. There is no way to pass structured metadata alongside prompts.

**Location**: `textpolicy/environment/text_generation.py`, line 741

```python
reward = self.reward_fn(
    prompt=self.current_prompt,
    completion=response_text,
    example={},  # <-- HARDCODED EMPTY
    tokenizer=self.tokenizer,
    truncated=truncated
)
```

## What Works

- Custom reward functions via `@tp.reward` decorator
- Reward receives `prompt` and `completion` correctly
- All 31 tests pass
- Basic training loop runs

## What Doesn't Work

Cannot use reward functions that need additional context per example:

```python
@tp.reward
def sql_execution_reward(prompt, completion, example, **kwargs):
    db_id = example['db_id']        # KeyError - example is {}
    gold_sql = example['gold_sql']  # KeyError - example is {}
    return evaluate(db_id, completion, gold_sql)
```

## Use Case

Text-to-SQL training with Spider benchmark. Each training example needs:
- `prompt`: Schema + natural language question
- `example['db_id']`: Which database to execute against
- `example['gold_sql']`: Ground truth SQL for comparison

## Litmus Test

This should work:

```python
examples = [
    {"db_id": "concert_singer", "gold_sql": "SELECT COUNT(*) FROM singer"},
    {"db_id": "pets_1", "gold_sql": "SELECT COUNT(*) FROM pets"}
]
prompts = [
    "Schema: singer(id, name)\nQuestion: How many singers?",
    "Schema: pets(id, name)\nQuestion: How many pets?"
]

@tp.reward
def check_example(prompt, completion, example, **kwargs):
    print(f"db_id: {example.get('db_id')}")  # Should print actual db_id
    return 1.0

env = TextGenerationEnv(prompts, check_example, examples=examples, tokenizer=tok)
env.reset()
env.step(some_action)
# Reward function should receive example={'db_id': 'concert_singer', 'gold_sql': '...'}
```

## Proposed Fix

Add optional `examples` parameter to `TextGenerationEnv`:

```python
def __init__(
    self,
    prompts: List[str],
    reward_fn: Callable[[str, str, dict], float],
    max_tokens: int = 25,
    seed: int = 42,
    tokenizer: Any = None,
    examples: Optional[List[dict]] = None  # NEW
):
    self.prompts = prompts
    self.examples = examples or [{} for _ in prompts]  # Default to empty dicts
    ...
```

Then in `step()`:

```python
prompt_index = (self.current_episode - 1) % len(self.prompts)
current_example = self.examples[prompt_index]

reward = self.reward_fn(
    prompt=self.current_prompt,
    completion=response_text,
    example=current_example,  # Pass actual example
    ...
)
```

## Acceptance Criteria

- [ ] `TextGenerationEnv` accepts optional `examples: List[dict]` parameter
- [ ] When provided, `examples[i]` is passed to reward function for `prompts[i]`
- [ ] When not provided, behavior unchanged (empty dict)
- [ ] Litmus test above passes

**Comment by teilomillet** [comment]
Fixed in commit f8853f8. Verified: example metadata now passes through to reward functions.
